### ISSUE TITLE
estimating MS-DSGE models

### ISSUE THREAD

**Issue by Pedro0304** [jmaih/RISE_toolbox#24]
Hi Junior,

Is there any example to estimate MS-DSGE in the toolbox? because I haven't found any, it seems that all of them uses calibrated parameters and when I try to use the estimate() function in a MS-DSGE models i get errors "parameter rho_x is not controlled by const", for example. I know that to calculate the likelihood for the estimation is with a different filter than the Kalman filter but I do not know if one has to implement that option in the estimate() function or so.

Thanks a lot and regards.
Pedro.

**Comment by jmaih** [comment]
Hi Pedro,

This just a syntax error. You most likely declared "rho_x" to be governed by some Markov chain and when it came to parameterizing it you forgot to put the name of the Markov chain in parentheses. 

For example the code below will issue an error
```
endogenous X Y Z
parameters alpha beta, gamma
parameters(policy,2) rho_x
model
X=Y*log(Z);
...

parameterization
alpha, 0.7;
beta, 0.99;
gamma, 1;
rho_x, 0.85;
```

This is because `rho_x` is declared to be controlled by Markov chain `policy` but then in the parameterization block, we treat it in the same way as alpha, beta and gamma. The correct way to go about is to rewrite the parameterization block as follows:

```
parameterization
alpha, 0.7;
beta, 0.99;
gamma, 1;
rho_x(policy,1), 0.85;
rho_x(policy,2), 0.65;
```
Let me know if this helps,

Cheers,

Junior

**Comment by Pedro0304** [comment]
Thanks a lot for answered me really fast, unfortunately the problem is not that. That problem of syntaxis was already fixed by me before. 

To be specific, what i'm doing is trying to run the code "master.m" in the directory "C:\...\RISE_toolbox-master\examples\VariousModels\PeterIreland\NKperspective_JMCB2011" but instead of using the code "nkp.rs" I'm using "nkp_switch.rs" because i want to estimate the MS-DSGE model and the code "master_switch.m" just solves the MS-DSGE model with calibration NOT ESTIMATES IT. 

So to sum up I replace the line "m=rise('nkp');" in "master.m" with the line "m=rise('nkp_switch');" and i got that error, what i can infer is that the problem comes from the file "create_parameters.m" because now i should define two priors for the two states of the variable and also for the transition probabilities but the problem is that i don't know which syntax i should use.

Do you have an example which estimates a MS-DSGE NOT CALIBRATES IT? I haven't found any in the RISE toolbox

Thanks a lot and best,
Pedro.

**Comment by Pedro0304** [comment]
All the errors are shown below:

**_Error using generic_tools.parameter_position_and_regimes/retrieval_engine (line 62)
parameter rho_x is not controlled by const

Error in generic_tools.parameter_position_and_regimes (line 32)
    [position(ipar),regime_states{ipar}]=retrieval_engine(pnames{ipar});

Error in generic/setup_calibration (line 82)
    [position,regime_states]=generic_tools.parameter_position_and_regimes(pnames,...

Error in dsge/setup_calibration (line 46)
obj=setup_calibration@generic_switch(obj,Calibration);

Error in generic/set/set_one_at_a_time (line 130)
            obj=setup_calibration(obj,propval);

Error in generic/set (line 116)
        set_one_at_a_time();

Error in dsge/set (line 105)
obj=set@generic_switch(obj,varargin{:});

Error in master (line 14)
m=set(m,'parameters',p);_**

**Comment by jmaih** [comment]
Hi Pedro,

That example was never meant for estimation. I recently added it to answer a question by a user who, I thought, needed to run some historical decomposition.

I you want to see an example with estimation, take a look at \examples\VariousModels\TaoZha\Tutorials\DSGE\Tutorial1 

In there you will find examples of estimation of dsge models with regime switching. You will still need to set up your own problem but hopefully this can help you started. 

Cheers,

Junior

**Comment by Pedro0304** [comment]
Hi Junior,

Thanks a lot! That’s what i wanted since the begining but i thought it wasn’t in the toolbox, I should have checked more carefully.
I really appreciate yor help.

Kind regards.
Pedro.

**Comment by addsam004** [comment]
Hi Junior
how do I modify the file \examples\VariousModels\TaoZha\Tutorials\DSGE\Tutorial1 to add exchange rate to the feedback rule

**Comment by addsam004** [comment]
Dear Junior 
Where can a find a guide to specifying observation equations for the MS-dsge model. I have to write mine own equations but is becoming more confusing for me.

Best,
Sam

**Comment by jmaih** [comment]
Hi Sam,

1. How to modify ... to add exchange rate to the feedback rule: Do you have a model with exchange rate? The equations of your model should be the ones that you code up in RISE. You do not just add a new variable in the system without providing an extra equation.

1. There is no need to specify observation equations. All you have to do is to list the observable variables, RISE will figure out what the measurement equations should look like.

Cheers,

Junior

**Comment by addsam004** [comment]
Thank you will work on it.

Best,
Sam

**Comment by addsam004** [comment]
Version Name: 6.0.2, Version Code: 124w.wr33I.'ren33sse3t3as3a
